python/fledge/services/core/api/proxy.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+# FLEDGE_BEGIN
+# See: http://fledge-iot.readthedocs.io/
+# FLEDGE_END
+import json
+import logging
+import aiohttp
+
+from aiohttp import web
+from fledge.common import logger
+from fledge.services.core import server
+from fledge.services.core.service_registry import exceptions as service_registry_exceptions
+from fledge.services.core.service_registry.service_registry import ServiceRegistry
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2022 Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+_logger = logger.setup(__name__, level=logging.INFO)
+SVC_TYPE = "BucketStorage"
+
+
+def setup(app):
+    app.router.add_route('*', '/fledge/bucket{head:.*}', handler)
+
+
+async def handler(request):
+    """ For example: For a bucket service
+        POST -   /bucket
+        GET -    /bucket/([0-9][0-9]*)$
+        DELETE - /bucket/([0-9][0-9]*)$
+        PUT -    /bucket/([0-9][0-9]*)$
+        PUT -    /bucket/match
+    """
+    allow_methods = ["GET", "POST", "PUT", "DELETE"]
+    if request.method not in allow_methods:
+        raise web.HTTPMethodNotAllowed(method=request.method, allowed_methods=allow_methods)
+    try:
+        data = await request.json() if request.method != 'GET' else None
+        svc, token = await _get_service_record_info_along_with_bearer_token()
+        if svc._name not in server.Server._API_PROXIES.keys():
+            raise web.HTTPForbidden()
+        url = str(request.url).split('fledge/')[1]
+        result = await _call_microservice_service_api(request.method, svc._protocol, svc._address, 
+                                                      svc._port, url, token, data)
+    except Exception as ex:
+        msg = str(ex)
+        raise web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
+    else:
+        return web.json_response({"info": result})
+
+
+async def _get_service_record_info_along_with_bearer_token():
+    try:
+        service = ServiceRegistry.get(s_type=SVC_TYPE)
```

Comment:
We should not be testing service type for the reasons noted above.

Reply:
This was added only to get the info for a service (like service port, address, bearer token) so that Admin API -> forward those calls to microservice API.

Are we going to pass service name or any identifier in request header for public API exposed endpoints?

For example:  To achieve below we require identifier related to microservice

curl -sX GET localhost:8081/foglamp/bucket/1   => GET <MICROSVC_PROTOCOL>://<MICROSVC_ADDRESS>:<MICROSVC_PORT>/bucket/1

OR Am I missing?

Reply:
We pass the service name when we setup the proxy, that should be enough to get the information you need I believe